addOns/spider/src/main/java/org/zaproxy/zap/extension/spider/SvgHrefSpider.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.spider;
+
+import java.io.ByteArrayInputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.regex.Pattern;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import net.htmlparser.jericho.Source;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.network.HttpMessage;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXParseException;
+import org.zaproxy.zap.spider.parser.SpiderParser;
+import org.zaproxy.zap.utils.XmlUtils;
+
+public class SvgHrefSpider extends SpiderParser {
+    private static final Logger LOGGER = LogManager.getLogger(SvgHrefSpider.class);
+    private static final Pattern PATTERN_SVG_EXTENSION =
+            Pattern.compile("\\.svg\\z", Pattern.CASE_INSENSITIVE);
+    private static final String HREF_EXPRESSION = "//*[@href or @HREF]";
```

Comment:
There doesn't seem to be a way to do case insensitive xpath expressions. So if a site/app happens to mix case it'll be missed, but that is hopefully a nearly non-existent case. This also has to be handled down around L130.